src/common/com/intellij/plugins/haxe/util/HaxePsiUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2015 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.util;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.plugins.haxe.lang.psi.HaxeIdentifier;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiWhiteSpace;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HaxePsiUtils {
+  static public <T extends PsiElement> T getAncestor(PsiElement element, Class<T> clazz) {
+    if (element == null) return null;
+    if (clazz.isAssignableFrom(element.getClass())) return (T)element;
+    return getAncestor(element.getParent(), clazz);
+  }
+
+  static public <T extends PsiElement> T getChild(PsiElement element, Class<T> clazz) {
+    if (element == null) return null;
+    for (PsiElement psiElement : element.getChildren()) {
+      if (clazz.isAssignableFrom(psiElement.getClass())) return (T)psiElement;
+    }
+    return null;
+  }
+
+  static public PsiElement getToken(PsiElement element, String token) {
+    for (ASTNode node : element.getNode().getChildren(null)) {
+      if (node.getText().equals(token))  return node.getPsi();
+    }
+    return null;
+  }
+
+  static public <T extends PsiElement> T getChild(PsiElement element, Class<T> clazz, String text) {
```

Comment:
I changed it to getChildWithText. Semantically it is just text. It can be a name, a keyword a random text. You can change it later if you think it is better getNamedChild.